Program.cs

```diff
@@ -9,6 +11,8 @@
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
+builder.Services.AddSingleton<ITodoRepository, TodoRepository>();

```

Comment:
Wait, so it just uses generic types for DI? That's really cool!

Reply:
Yes, thats why I'm so confused by Swift, this is how nice it is in .NET

However if you need more control over object creation you can provide it a `ServiceProvider -> T` callback like Swinject

Reply:
I think part of the difference is a limitation of Swift's type system -- there's no way to inspect the type signature of `init` to see what arguments it takes.

There is one thing I can think of that might still be a problem in C\#: How does this handle a class with multiple constructors?